text/0000-interactive-app-creation.md

```diff
@@ -0,0 +1,142 @@
+- Start Date: 2020-06-12
+- Relevant Team(s): ember-cli, framework, learning
+- RFC PR: (after opening the RFC PR, update this with a link to it and update the file name)
+- Tracking: (leave this empty)
+
+# Interactive New Ember App Creation
+
+## Summary
+
+As part of the effort to make new Ember apps more conformant for digital accessibility requirements at a global scale, this RFC proposes an interactive workflow for new Ember apps. This will also have the benefit of assisting new users who prefer an interactive model of new app creation. 
```

Comment:
> As part of the effort to make new Ember apps more conformant for digital accessibility requirements at a global scale, this RFC proposes an interactive workflow for new Ember apps.

What's the reasoning behind this sentence?

How does an interactive app creation flow help "accessibility requirements at a global scale", what does it mean concretely?

What's the tax/cost for maintaining two flows for creating apps (interactive and non-interactive)? I don't know the answer to this, but someone on the ember-cli team should know. Do we want to take on this tax?

How does an interactive flow make things easier compared to a non-interactive, for a blind/deaf/etc person?

How does other terminal software, editors, etc. work? To effectively develop applications there are usually a ton of other tooling needed. Maybe these problems are better solved at a higher level of abstraction, i.e. at the [editor](https://github.com/microsoft/vscode/issues/88987), [terminal](https://github.com/tspivey/tdsr) and [os](https://www.apple.com/accessibility/mac/) level?

For other parts of Ember, we often contemplate whether a problem is best solved within the Ember project, or outside. For example, the upcoming changes to the build system (Embroider) are a lot about piggy-backing on existing tooling in the JS ecosystem, instead of building everything ourselves.

Reply:
> How does an interactive app creation flow help "accessibility requirements at a global scale", what does it mean concretely?

Agree with this critique of the wording. The interactive flow just makes it easier to use the new `--lang` option without remembering it exists, AFAICT that is the only way this relates to A11Y.

> What's the tax/cost for maintaining two flows for creating apps (interactive and non-interactive)? I don't know the answer to this, but someone on the ember-cli team should know. Do we want to take on this tax?

The nice thing here is that **everything** here desugars into command line options that already exist. This means there isn't a duplicated cost of each option/etc, and the only new cost is the wizard / prompting itself. All of the underlying features are already implemented (or in the process of being implemented WRT to `--lang`) and tested.

From an ember-cli team perspective, I think this RFC is a very very large win for simplicity (both ember-cli's codebase, and ember-cli's users).

Reply:
That it desugars into existing (or future) command line options sounds good. Sort of like the `.ember-cli` file which afaik share much code with the CLI flags.

I don't know enough about ember-cli to determine whether the implementation cost is worth it. My point above was mainly to ask a few (uncomfortable) questions regarding prioritisation, and it's up to you on the core team to decide of course.